modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -196,7 +200,7 @@ public void processOpts() {
             apiDocTemplateFiles.remove("api_doc.mustache");
         }
 
-        if (!("feign".equals(getLibrary()) || "resttemplate".equals(getLibrary()) || usesAnyRetrofitLibrary())) {
+        if (!("feign".equals(getLibrary()) || "resttemplate".equals(getLibrary()) || usesAnyRetrofitLibrary() || "google-api-client".equals(getLibrary()))) {
```

Comment:
oh I see other library has done it too... @wing328, is this normal? Since we are using objects, we should'nt have any "if" in the JavaClientCodegen... It should all be in specific file of each generator...
Or there is a good reason for this ??

Reply:
Yeah, it appears there are no subclasses of `JavaClientCodegen`, see above comment, @wing328 probably has a more complete explanation.

Reply:
Ok. It's weird because for example, for the server generator "java play framework" which I maintain, there is: 
`JavaPlayFrameworkCodegen` -> `AbstractJavaCodegen` -> `DefaultCodegen`. 
This prevent having "if" everywhere.

Same thing for Typescript jQuery which I maintain too:
`TypeScriptJqueryClientCodegen` -> `AbstractTypeScriptClientCodegen` -> `DefaultCodegen`

I wonder why the JavaClientCodegen are all mixed in one file. 
For the moment, don't change anything. We will see with @wing328 answers :)

Reply:
That makes sense. Yeah, a refactor into subclasses probably belongs in a separate pull request, but I'm game to do whatever you suggest.

Reply:
This is ok for now and belong in another pull request like you said. :)

Reply:
When we added support for other HTTP libraries, we didn't foresee we can support so many libraries (9 HTTP libraries support so far thanks to the awesome community). I agree a separate PR to refactor into subclasses would be ideal.

Reply:
That makes sense. Is this waiting on any additional reviews? Let me know if any more changes are needed here. Thanks! This will really help out our company in our migration to use Swagger clients for all of our service-to-service communication. (And possibly we could use this for our Android app, too).

Reply:
@charlescapps I'll take another look later today and merge accordingly.